crates/rolldown/src/module_loader/normal_module_task.rs

```diff
@@ -127,6 +142,26 @@ impl NormalModuleTask {
         let analyzed_side_effects = stmt_infos.iter().any(|stmt_info| stmt_info.side_effect);
         DeterminedSideEffects::Analyzed(analyzed_side_effects)
       });
+
+    let default_side_effects = self
+      .package_json
+      .as_ref()
+      .and_then(|p| {
+        p.check_side_effects_for(&stable_resource_id).map(DeterminedSideEffects::PackageJson)
+      })
+      .unwrap_or_else(|| {
+        let analyzed_side_effects = stmt_infos.iter().any(|stmt_info| stmt_info.side_effect);
+        DeterminedSideEffects::Analyzed(analyzed_side_effects)
+      });
+
+    let side_effects = match hook_side_effects {
+      Some(side_effects) => match side_effects {
+        HookSideEffects::True => default_side_effects,
+        HookSideEffects::False => DeterminedSideEffects::PackageJson(false),
+        HookSideEffects::NoTreeshake => unimplemented!(),
```

Comment:
@IWANABETHATGUY Here need to disable treeshake for the module, look like it need to implement it.

Reply:
I looked into the feature 'no-treeshake'. If it's only for compatibility and no solid reasons, there's no need to implement the feature for adding extra complexity.

Reply:
I searched around and found the only real-world use case if for `lottie-web` which used `eval`. https://github.com/rollup/rollup/issues/2963

Reply:
It's not hard to implement this feature, we just need to mark all `StmtInfo` with `side_effect: true`. But still, if it is a overengineering feature, it's not worth for us to take actions on it.

Reply:
Yeah.It not be hurry to implement it. Here common case is `false` or `undefined`

Reply:
Saw that `unimplemented!` macro, let's keep it for a while.